unittests/test_config.py

```diff
@@ -12,10 +12,34 @@
 
 
 
+@pytest.fixture(params=['full', 'parts'])
+def site_config(request):
+    # `unittests/resources/config/settings.py` should be equivalent to loading
+    # the `unittests/resources/config/settings-part*.py` files
+    if request.param == 'full':
+        return config.load_config('unittests/resources/config/settings.py')
+    else:
+        return config.load_config(
+            'unittests/resources/config/settings-part1.py',
+            'unittests/resources/config/settings-part2.py',
+            'unittests/resources/config/settings-part3.py'
+        )
+
+
 def test_load_config_python():
     config.load_config('reframe/core/settings.py')
 
 
+def test_load_multiple_configs():
+    site1 = config.load_config('unittests/resources/config/settings.py')
+    site2 = config.load_config(
+        'unittests/resources/config/settings-part1.py',
+        'unittests/resources/config/settings-part2.py',
+        'unittests/resources/config/settings-part3.py'
+    )
+    assert site1._site_config == site2._site_config
```

Comment:
Unit tests should assert API behaviour. The `_site_config` is an implementation detail and should not be read like that... Better add a couple for assertions for the things that you expect to have changed.